File: Picturra.com/Scripts/Profile/Project.js
var ProjectViewModel = function (id, title, medium, height, width, year, likeCount, images) {
    var self = this;
    self.Id = id;
    self.uniqueId = guid();
    self.Title = ko.observable(title);
    self.Medium = ko.observable(medium);
    self.Height = ko.observable(height);
    self.Width = ko.observable(width);
    self.Year = ko.observable(year);
    self.LikeCount = ko.observable(likeCount);

    self.Images = ko.observableArray(ko.utils.arrayMap(images, function(image) {
        return new ImageViewModel(image.Id, image.Url, image.Caption, image.PublicKey, image.Name, guid());
    }));

    self.AddImage = function () {
        if (self.Images().length < 3) {
            self.Images.push(new ImageViewModel('', '', '', '', ''));
        }
    }

    self.RemoveImage = function (image) {
        self.Images.remove(image);
    }
}

File: Picturra.Models/Profile/Project.cs
using System.Collections.Generic;
namespace Picturra.Models.Profile
{
    public class Project : IModelBase<int>
    {
        public Project()
        {
            TitleImage = new Image.Image();
            Images = new List<Image.Image>();
        }
        public int Id { get; set; }
        public Image.Image TitleImage { get; set; }
        public string Title { get; set; }
        public Enums.Enums.Medium Medium { get; set; }
        public int Height { get; set; }
        public int Width { get; set; }
        public string Year { get; set; }
        public int LikeCount { get; set; }
        public List<Image.Image> Images { get; set; }
    }
}